magma/lib/magma/attributes/matrix.rb

```diff
@@ -31,6 +33,18 @@ def revision_to_payload(record_name, new_value, loader)
       [ name, new_value ]
     end
 
+    def expand(matrix_value)
+      # Provide an embedded data frame inside of the TSV
+      #   that includes the selected columns names + values.
+
+      # matrix_value.to_json is actually a JSON string.
+      JSON.parse(matrix_value.to_json)
```

Comment:
There, might be `as_json` which would return the json compatible structure itself pre stringifying.  It's not standard but many apis do this.  If not, this works.

Reply:
Unfortunately seems like not included.